src/System.IO.FileSystem.DriveInfo/tests/DriveInfo.Windows.Tests.cs

```diff
@@ -0,0 +1,194 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Xunit;
+using System.Text;
+
+namespace System.IO.FileSystem.DriveInfoTests
+{
+    public class DriveInfoTests
+    {
+        [Fact]
+        public void TestConstructor()
+        {
+            string[] invalidInput = { ":", "://", @":\", ":/", @":\\", "Az", "1", "a1", "\\share", "\\", "c ", string.Empty, " c" };
+            string[] variableInput = { "{0}", "{0}", "{0}:", "{0}:", @"{0}:\", "{0}:\\", "{0}://" };
+
+            // Test Invalid input
+            foreach (var input in invalidInput)
+            {
+                Assert.Throws<ArgumentException>(() => { new DriveInfo(input); });
+            }
+
+            // Test Null
+            Assert.Throws<ArgumentNullException>(() => { new DriveInfo(null); });
+
+            // Test Valid DriveLetter
+            var validDriveLetter = GetValidDriveLettersOnMachine().First();
+            foreach (var input in variableInput)
+            {
+                string name = string.Format(input, validDriveLetter);
+                DriveInfo dInfo = new DriveInfo(name);
+                Assert.Equal(string.Format(@"{0}:\", validDriveLetter), dInfo.Name);
+            }
+        }
+        
+        [Fact]
+        public void TestGetDrives()
+        {
+            var validExpectedDrives = GetValidDriveLettersOnMachine();
+            var validActualDrives = DriveInfo.GetDrives();
+
+            // Test count
+            Assert.Equal(validExpectedDrives.Count(), validActualDrives.Count());
+
+            for (int i = 0; i < validActualDrives.Count(); i++)
+            {
+                // Test if the driveletter is correct
+                Assert.True(validExpectedDrives.Contains(validActualDrives[i].Name[0]));
+            }
+        }
+
+        [Fact]
+        public void TestDriveFormat()
+        {
+            var validDrive = DriveInfo.GetDrives().Where(d => d.DriveType == DriveType.Fixed).First();
+            const int volNameLen = 50;
+            StringBuilder volumeName = new StringBuilder(volNameLen);
+            const int fileSystemNameLen = 50;
+            StringBuilder fileSystemName = new StringBuilder(fileSystemNameLen);
+            int serialNumber, maxFileNameLen, fileSystemFlags;
+            bool r = GetVolumeInformation(validDrive.Name, volumeName, volNameLen, out serialNumber, out maxFileNameLen, out fileSystemFlags, fileSystemName, fileSystemNameLen);
+            var fileSystem = fileSystemName.ToString();
+            if (r)
+            {
+                if (fileSystem != validDrive.DriveFormat)
+                {
+                    Assert.False(true, string.Format("Error, unexpected result returned: {0}, Expected: {1}", validDrive.DriveFormat, fileSystem));
```

Comment:
I think it'd be clearer to just do `Assert.Equal(fileSystem, validDrive.DriveFormat)`, then you can get rid of the if as well.